SDWebImage/Core/SDAnimatedImageView.m

```diff
@@ -129,12 +129,12 @@ - (void)setImage:(UIImage *)image
     
     // We need call super method to keep function. This will impliedly call `setNeedsDisplay`. But we have no way to avoid this when using animated image. So we call `setNeedsDisplay` again at the end.
     super.image = image;
-    if ([image.class conformsToProtocol:@protocol(SDAnimatedImage)]) {
+    if ([image.class conformsToProtocol:@protocol(SDAnimatedImage)] && [(id<SDAnimatedImage>)image animatedImageFrameCount] > 1) {
```

Comment:
For some case, like a single frame `GIF` in `SDAnimatedImage`, this will cause behavior changes. But I think still worth.

Single frame animated image format (1 frame GIF) actually can always consider as `Static image`, right ?